components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/SQLQueries.java

```diff
@@ -374,16 +388,32 @@
             "(ACCESS_TOKEN, REFRESH_TOKEN, CONSUMER_KEY_ID, AUTHZ_USER, TENANT_ID, USER_DOMAIN, TIME_CREATED, " +
             "REFRESH_TOKEN_TIME_CREATED, VALIDITY_PERIOD, REFRESH_TOKEN_VALIDITY_PERIOD, TOKEN_SCOPE_HASH, " +
             "TOKEN_STATE, USER_TYPE, TOKEN_ID, GRANT_TYPE, SUBJECT_IDENTIFIER, ACCESS_TOKEN_HASH, REFRESH_TOKEN_HASH," +
-            " IDP_NAME) SELECT ?,?,ID,?,?,?," + "?,?,?,?,?,?,?,?,?," + "?,?,? FROM IDN_OAUTH_CONSUMER_APPS WHERE " +
-            "CONSUMER_KEY=?";
+            "IDP_ID) SELECT ?,?,ID,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,IDP.ID FROM IDN_OAUTH_CONSUMER_APPS, IDP WHERE " +
+            "CONSUMER_KEY=? AND IDP.NAME=? AND IDP.TENANT_ID=?";
```

Comment:
Do we have an index for IDNP

Reply:
Will look into this. I have not checked on the indexes for any of the new additions.

Reply:
In the IDP table, there is an index for NAME + TENANT_ID created by their unique constraint